### DISCUSSION TITLE
Maximum update depth exceeded when extracting usePopper into custom hook.

### DISCUSSION THREAD

**Discussion by ajkl2533** [floating-ui/floating-ui#1395] [Q&A]
Hi, I want to create a custom hook that should encapsulate popper with the default settings and then it can be reused in multiple places. 
My plan was that the hook accepts only the reference and then returns React component. Unfortunately, I hit an issue since setting the popper ref is causing an infinite loop that ends with the "Maximum update depth exceeded." error. I'm pretty sure that is just a matter of memorization in the correct place, but I can't find that place.

I will be glad for any advice, thank you.

Codesandbox example: https://codesandbox.io/s/react-popper-poc-qqxm4?file=/src/Hook.js

**Comment by palpich** [comment]
This should help: https://popper.js.org/react-popper/v2/faq/#why-i-get-render-loop-whenever-i-put-a-function-inside-the-popper-configuration
